### PR TITLE
executable polish

### PR DESCRIPTION
**Pull request by dimalinux**
* Fixed the signal handler
* Put the executable in a same-named directory
* Added a version
* Makefile build target now uses `go install` for the version handling

I was thinking we may want to launch this executable as part of our integration test setup in the `atomic-swap` repository instead of calling the code directly. Since this executable builds easily using `go install`, it wouldn't be hard to add.

### CODE REVIEW

File: cmd/relayer/main.go [hunk 1]
```diff
@@ -102,7 +102,7 @@ func setLogLevels(c *cli.Context) error {
 		return fmt.Errorf("invalid log level %q", level)
 	}
 
-	_ = logging.SetLogLevel("cmd", level)
```
Comment (dimalinux) [inline cmd/relayer/main.go:105]: I wasn't sure what string to put here. It's in the `main` package located in the `cmd/relayer` folder. There's a package named `relayer`, so I couldn't use that.

File: cmd/relayer/main.go [hunk 2]
```diff
@@ -171,11 +173,11 @@ func run(c *cli.Context) error {
 		return err
 	}
 
-	sigc := make(chan os.Signal, 1)
-	signal.Notify(sigc, syscall.SIGINT, syscall.SIGTERM)
-	defer signal.Stop(sigc)
```
Comment (dimalinux) [inline cmd/relayer/main.go:176]: This version wasn't canceling the context.

File: Makefile [hunk 3]
```diff
@@ -9,4 +9,5 @@ test:
 	go test ./...
 
 build:
-	cd cmd && go build -o ../bin/relayer 
\ No newline at end of file
+	mkdir -p bin
+	GOBIN="$(CURDIR)/bin/" go install ./cmd/...
```
Comment (noot) [inline Makefile:13]: why `go install` vs `go build`? don't need to change this, just curious about the reasoning/difference

**Review by noot — APPROVED** [review]
looks good!

File: Makefile (hunk 3 above)
Comment (dimalinux) [reply Makefile:13]: Once we put a release tag on the repo, you can use the tag as part of the `go install` to install a specific release. For this line specifically, we are installing off of the latest commit, not a tag.  If this latest commit is tagged, the version tag should show up in our version sting regardless. We can verify that when we do our first tag.

**Pull request merged**